src/buffer/buffer_map.rs

```diff
@@ -311,9 +393,49 @@ impl Accessed for BufferMap {
     }
 }
 
+impl<T: 'static + Send + Sync> JoinedValue for Vec<T> {
+    type Buffers = Vec<Buffer<T>>;
+}
+
+impl<B: 'static + Send + Sync + AsAnyBuffer + Clone> BufferMapLayout for Vec<B> {
+    fn try_from_buffer_map(buffers: &BufferMap) -> Result<Self, IncompatibleLayout> {
+        let mut downcast_buffers = Vec::new();
+        let mut compatibility = IncompatibleLayout::default();
+        for i in 0..buffers.len() {
+            if let Ok(downcast) = compatibility.require_buffer_by_index::<B>(i, buffers) {
+                downcast_buffers.push(downcast);
+            }
+        }
+
+        compatibility.as_result()?;
+        Ok(downcast_buffers)
+    }
+}
+
+impl<T: 'static + Send + Sync, const N: usize> JoinedValue for SmallVec<[T; N]> {
+    type Buffers = SmallVec<[Buffer<T>; N]>;
+}
+
+impl<B: 'static + Send + Sync + AsAnyBuffer + Clone, const N: usize> BufferMapLayout
+    for SmallVec<[B; N]>
+{
+    fn try_from_buffer_map(buffers: &BufferMap) -> Result<Self, IncompatibleLayout> {
+        let mut downcast_buffers = SmallVec::new();
+        let mut compatibility = IncompatibleLayout::default();
+        for i in 0..buffers.len() {
+            if let Ok(downcast) = compatibility.require_buffer_by_index::<B>(i, buffers) {
+                downcast_buffers.push(downcast);
+            }
+        }
+
+        compatibility.as_result()?;
+        Ok(downcast_buffers)
+    }
+}
+
 #[cfg(test)]
 mod tests {
-    use crate::{prelude::*, testing::*, BufferMap};
+    use crate::{prelude::*, testing::*, AddBufferToMap, BufferMap};
```

Comment:
Should `AddBufferToMap` be in prelude?

Reply:
I'm a bit torn on this and open to opinions. I'm leaning towards no because realistically I think it'll only be used in tests and maybe in the diagram module. I think it's very unlikely that downstream users will ever use a `BufferMap` directly.